### PR TITLE
Added support for SLES, concat fragments in rules file, and ability to unmanage and stop service

### PR DESCRIPTION
**Pull request by exodusftw**

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -380,16 +380,34 @@
...
-      enable    => true,
-      hasstatus => true,
-      restart   => $service_restart,
-      stop      => $service_stop,
-      subscribe => [ File['/etc/audit/auditd.conf'], Concat['audit-file'] ],
-    }
+    $service_ensure = 'running'
+    $service_enable = true
+  }
+  else {
```
Comment (kemra102) [inline manifests/init.pp]: We don't need an else here. If the service is not managed it is assumed that it is either being managed manually or by some other process. This statement would stop that from being a possibility.

File: templates/audit.rules.begin.fragment.erb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# Default Rule - Delete ALL
+-D
+
+# Set Buffer size - increase for Busy Systems
+-b 8192
```
Comment (kemra102) [inline templates/audit.rules.begin.fragment.erb]: Having a beginning fragment is fine but setting the buffer size like this is a little too opinionated for my tastes. It doesn't give others the opportunity to tune it to their requirements.

File: manifests/init.pp [hunk 3]
```diff
@@ -380,16 +380,34 @@
...
+    target  => $rules_file,
+    content => template('auditd/audit.rules.begin.fragment.erb'),
+    order   => '00'
+  }
+
+  concat::fragment{ 'auditd_rules_end':
+    target  => $rules_file,
+    content => template('auditd/audit.rules.end.fragment.erb'),
+    order   => '99'
+  }
```
Comment (kemra102) [inline manifests/init.pp]: Probably don't need this, the whole file is puppet managed so having a footer that marks the end of Puppet managed rules is redundant.

**Comment by kemra102** [comment]
Thanks for the pull request, nice to have someone on the SUSE side looking at this as I have no access to any SLES systems right now.

I have a few comments I've added in-line that I'm happy to discuss further.

File: manifests/init.pp (hunk 1 above)
Comment (exodusftw) [inline manifests/init.pp]: Personally - I would say if you're going to manage the package, config, etc. - it makes sense to also manage the service and ensure it's running/stopped.

That being said - I went ahead and left the manage_service boolean intact with the service resource wrapped in the "if" clause - and just set the ensure/enable attributes as parameters that would only be realized if the service is being managed - either running/stopped. Hopefully that's a reasonable compromise.

File: templates/audit.rules.begin.fragment.erb (hunk 2 above)
Comment (exodusftw) [inline templates/audit.rules.begin.fragment.erb]: Can't argue - I went ahead and set the default as 8192 and parameterized the value - if you still feel that 8192 is too large a default value - let me know and I'll drop it down to something tiny like 128

File: manifests/init.pp (hunk 3 above)
Comment (exodusftw) [inline manifests/init.pp]: that's reasonable - I removed the ending fragment

**Comment by kemra102** [comment]
The compromise is the perfect, best of both worlds in my opinion, pull looks great, many thanks!

**Pull request merged**